articles/group-operator/group.md

```diff
@@ -0,0 +1,141 @@
+
+So $group operator also works in a similar fashion. Let’s see its syntax in order to make this more clear :
+
+`{ $group: { _id: <expression>, <field1>: { <accumulator1> : <expression1> }, ... } }`
+
+An _id field in the output documents contains the distinct group by key. The output documents can also contain computed fields that hold the values of some accumulator expression grouped by the $group‘s _id field.
+
+Basically, the _id field here will take those fields by which you want to group the documents and it is compulsory to use _id here otherwise the operator won’t work.
+
+### Important points to remember
+
+- Can not be used with the LogReduce operator.
+
+- When parsing and naming (aliasing) fields, avoid using the names of grouping functions or other operators as field names.
+
+- When using count, or any grouping function, remember to include the underscore before the field name (sort by _count).
+
+- Multiple aggregation functions can be on the same line but you can't include another function, such as a math function, on the same line of a query.
+
+Now, let’s look at an example.
+
+Suppose we have a collection of 3 people with their names, gender, age and hobbies:
+
+```JavaScript
+[
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbd"),
+        "name" : "Harit Joshi",
+        "age" : 18,
+        "gender" : "Male",
+        "hobbies" : [
+                "sports",
+                "cooking",
+                "gaming"
+        ]
+    },
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbe"),
+        "name" : "Maria Swartz",
+        "age" : 27,
+        "gender" : "Female",
+        "hobbies" : [
+                "sports",
+                "swimming",
+                "gaming"
+        ]
+    },
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbf"),
+        "name" : "Billy Marton",
+        "age" : 34,
+        "gender" : "Male",
+        "hobbies" : [
+                "singing",
+                "cooking",
+                "jogging"
+        ]
+    }
+]
+```
+
+Let's say we want to find how many males and how many females are there in this document. In this short example, it is pretty clear that there are 2 males and 1 female.
+
+```js
+db.person.aggregate([ { $group: {_id: {gender: "$gender"}, totalPeople: {$sum: 1}} } ]).pretty()
+```
+
+Output:
+
+```JavaScript
+{ "_id" : { "gender" : "Female" }, "totalPeople" : 1 }
+{ "_id" : { "gender" : "Male" }, "totalPeople" : 2 }
+```
+
+Also, remember in order to make the $group operator work you must have to use an accumulator operator like $sum, $avg, $max, etc because as you know that the operator yields a combined result based on the grouping expression. Let's take another used case where this operator can shine. Suppose now we are interested in how many people have a similar hobby so how can we achieve that. Well it's not that much tough
```

Comment:
```suggestion
To make the $group operator work, you should use an accumulator operator like `$sum`, `$avg`, `$max`, etc since the operator yields a combined result based on the grouping expression. 

Let's take a look at another use case. Suppose now we are interested in how many people have a similar hobby,
```